pure.zsh

```diff
@@ -582,6 +583,29 @@ prompt_pure_setup() {
 	# to be available, it was added in Zsh 5.3.
 	autoload -Uz +X add-zle-hook-widget 2>/dev/null
 
+	# most used colors
+	typeset -gA prompt_pure_colors
+	prompt_pure_colors=(
```

Comment:
I would change this to 

```
	prompt_pure_colors=(
		path       blue
		branch     242
		git_arrow  cyan
		root       white
		user       242
		host       242
		exec_time  yellow
		virtualenv 242
		success    magenta
		error      red
  )
```

Just a small nit, but it makes more sense to me to have colors in the order that their sections appear (by default) on the prompt.

---

Additionally, I know it probably doesn't matter, but would you be opposed to moving the array to the top of the function? Considering there are things above it that currently reference the array, it makes more logical sense to have it be at the top despite it already being global. Would it make sense to you to do that in this situation?

Reply:
I will pass on both. The colors are sorted alphabetically because it's easier to insert them that way. The position of the array is after all the loading of all the external functions and before any actual function calls.